presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java

```diff
@@ -1070,43 +1072,89 @@ public void testViewAccessControl()
         assertAccessAllowed(
-                "SELECT * FROM test_nested_view_access",
-                privilege(getSession().getUser(), "test_view_access", CREATE_VIEW_WITH_SELECT_COLUMNS));
+                "SELECT * FROM test_nested_view_column_access",
+                privilege(getSession().getUser(), "test_view_column_access", CREATE_VIEW_WITH_SELECT_COLUMNS));
         assertAccessAllowed(
-                "SELECT * FROM test_nested_view_access",
-                privilege(getSession().getUser(), "test_view_access", SELECT_COLUMN));
+                "SELECT * FROM test_nested_view_column_access",
+                privilege(getSession().getUser(), "test_view_column_access", SELECT_COLUMN));
 
         // verify that INVOKER security runs as session user
         assertAccessAllowed(
                 viewOwnerSession,
-                "CREATE VIEW test_invoker_view_access SECURITY INVOKER AS SELECT * FROM orders",
+                "CREATE VIEW test_invoker_view_column_access SECURITY INVOKER AS SELECT * FROM orders",
                 privilege("orders", CREATE_VIEW_WITH_SELECT_COLUMNS));
         assertAccessAllowed(
-                "SELECT * FROM test_invoker_view_access",
+                "SELECT * FROM test_invoker_view_column_access",
                 privilege(viewOwnerSession.getUser(), "orders", SELECT_COLUMN));
         assertAccessDenied(
-                "SELECT * FROM test_invoker_view_access",
+                "SELECT * FROM test_invoker_view_column_access",
                 "Cannot select from columns \\[.*\\] in table .*.orders.*",
                 privilege(getSession().getUser(), "orders", SELECT_COLUMN));
 
         // change access denied exception to view
-        assertAccessDenied("SHOW CREATE VIEW test_nested_view_access", "Cannot show create table for .*test_nested_view_access.*", privilege("test_nested_view_access", SHOW_CREATE_TABLE));
-        assertAccessAllowed("SHOW CREATE VIEW test_nested_view_access", privilege("test_denied_access_view", SHOW_CREATE_TABLE));
+        assertAccessDenied("SHOW CREATE VIEW test_nested_view_column_access", "Cannot show create table for .*test_nested_view_column_access.*", privilege("test_nested_view_column_access", SHOW_CREATE_TABLE));
+        assertAccessAllowed("SHOW CREATE VIEW test_nested_view_column_access", privilege("test_denied_access_view", SHOW_CREATE_TABLE));
 
-        assertAccessAllowed(nestedViewOwnerSession, "DROP VIEW test_nested_view_access");
-        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_view_access");
-        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_invoker_view_access");
+        assertAccessAllowed(nestedViewOwnerSession, "DROP VIEW test_nested_view_column_access");
+        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_view_column_access");
+        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_invoker_view_column_access");
+    }
+
+    @Test
+    public void testViewFunctionAccessControl()
```

Comment:
That's too many test scenarios in single method. It's hard to follow what is actually happening.

Reply:
I would like to refactor these tests in: #3037

Currently it follows same pattern as `testViewColumnAccessControl`